src/lib/qnio/io_ssl.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Network IO library for VxHS QEMU block driver (Veritas Technologies)
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ *
+ * Contributions after 2014-08-15 are licensed under the terms of the
+ * GNU GPL, version 2 or (at your option) any later version.
+ */
+
+#include "defs.h"
+#include "qnio.h"
+
+static int
+read_ssl(struct endpoint *endpoint, void *buf, size_t len)
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_read(endpoint->ssl, buf, len));
+}
+
+static int
+write_ssl(struct endpoint *endpoint, const void *buf, size_t len)
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_write(endpoint->ssl, buf, len));
+}
+
+static int
+readv_ssl(struct endpoint *endpoint, struct iovec *vec, int count) 
+{
+    assert(endpoint->ssl != NULL);
+    return (SSL_read(endpoint->ssl, vec[0].iov_base, vec[0].iov_len));
```

Comment:
This is an bug waiting to happen.  The function doesn't look at the count argument and assumes the iovec has exactly one element.  I suggest dropping readv_ssl() and writev_ssl().